int16 state_x
int16 state_y
